util/crc32c.cc

```diff
@@ -340,8 +340,11 @@ static inline void Slow_CRC32(uint64_t* l, uint8_t const **p) {
   table0_[c >> 24];
 }
 
+#if defined(__x86_64__) || defined(_WIN64)
+__attribute__((target("sse4.2")))
```

Comment:
per https://clang.llvm.org/docs/AttributeReference.html#target and https://gcc.gnu.org/wiki/FunctionMultiVersioning , we need to define a "default" version of `Fast_CRC32()` if a non-default one is tagged, so the runtime can fallback to the default one at runtime if sse4.2 is not available on the target machine. so, probably we could just call `Slow_CRC32` in it?

i am concerned that older clang does not support "target" attribute, as it turns out this support was added in clang 3.7.0.

see https://releases.llvm.org/3.7.0/tools/clang/docs/AttributeReference.html and https://releases.llvm.org/3.6.0/tools/clang/docs/AttributeReference.html . search "gnu::target" in them.

and we claim that 

> We do depend on newer gcc/clang with C++11 support

at https://github.com/facebook/rocksdb/blob/master/INSTALL.md#compilation .

it seems that clang 3.3 already has decent C++11 support. per the table in https://clang.llvm.org/cxx_status.html .

when it comes to GCC, it had good support of C++11 since 4.8. see https://www.gnu.org/software/gcc/projects/cxx-status.html#cxx11

gcc supports the "target" function attribute since 4.5, so i guess we are safe. see https://gcc.gnu.org/onlinedocs/gcc-4.5.2/gcc/Function-Attributes.html .

so we might need something like https://github.com/facebook/rocksdb/pull/2950/files#diff-f54be34b9b64b37988070860edd87aafL341 ?